src/GraphQL.MicrosoftDI/ResolverBuilder.cs

```diff
@@ -0,0 +1,331 @@
+using System;
+using System.Threading.Tasks;
+using GraphQL.Builders;
+using GraphQL.Utilities;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.MicrosoftDI
+{
+    /// <summary>
+    /// A builder for field resolvers with no extra service types.
+    /// </summary>
+    public class ResolverBuilder<TSourceType, TReturnType>
+    {
+        private readonly FieldBuilder<TSourceType, TReturnType> _builder;
+        private bool _scoped;
+
+        /// <summary>
+        /// Initializes a new instance with the specified properties.
+        /// </summary>
+        public ResolverBuilder(FieldBuilder<TSourceType, TReturnType> builder, bool scoped)
+        {
+            _builder = builder;
+            _scoped = scoped;
+        }
+
+        /// <summary>
+        /// Specifies a type that is to be resolved via dependency injection during the resolver's execution.
+        /// </summary>
+        public ResolverBuilder<TSourceType, TReturnType, T1> WithService<T1>()
+            => new ResolverBuilder<TSourceType, TReturnType, T1>(_builder, _scoped);
+
+        /// <summary>
+        /// Specifies types that are to be resolved via dependency injection during the resolver's execution.
+        /// </summary>
+        public ResolverBuilder<TSourceType, TReturnType, T1, T2> WithServices<T1, T2>()
+            => new ResolverBuilder<TSourceType, TReturnType, T1, T2>(_builder, _scoped);
+
+        /// <inheritdoc cref="WithServices{T1, T2}"/>
+        public ResolverBuilder<TSourceType, TReturnType, T1, T2, T3> WithServices<T1, T2, T3>()
+            => new ResolverBuilder<TSourceType, TReturnType, T1, T2, T3>(_builder, _scoped);
+
+        /// <inheritdoc cref="WithServices{T1, T2}"/>
+        public ResolverBuilder<TSourceType, TReturnType, T1, T2, T3, T4> WithServices<T1, T2, T3, T4>()
+            => new ResolverBuilder<TSourceType, TReturnType, T1, T2, T3, T4>(_builder, _scoped);
+
+        /// <inheritdoc cref="WithServices{T1, T2}"/>
+        public ResolverBuilder<TSourceType, TReturnType, T1, T2, T3, T4, T5> WithServices<T1, T2, T3, T4, T5>()
+            => new ResolverBuilder<TSourceType, TReturnType, T1, T2, T3, T4, T5>(_builder, _scoped);
```

Comment:
I don't know how you feel about the proposal to add similar methods to all other builders. Their number will decrease with each new builder.

Reply:
I thought about it but figured they would either do one (specify all on one line) or the other (specify one per line) -- why bog down the API for both scenarios?

Reply:
Like would they really write:
```cs
graph.Field().Name("test")
    .Resolve()
    .WithService<T1>()
    .WithServices<T2, T3>()
    .Resolve((context, t1, t2, t3) => ...);
```
?  Just seemed like they would write `.WithServices<T1, T2, T3>`.  I can add the methods; it's not that hard.....if you think we should.  (A pile more tests, ugh.)

Reply:
Intellisense will help them write valid code, so we don't need to worry about that.

Reply:
So once they type `.WithService<T1>()`, the next line intellisense will only show them `.WithService` so they can't really screw it up.

Reply:
ok